cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/action/ActionContext.java

```diff
@@ -35,4 +42,31 @@
   @Override
   SettableArguments getArguments();
 
+  /**
+   * This method is based on the assumption that the plugin has an input schema or output schema and the DAG of the
+   * operations are known. Actions do not have input and output schema, and the DAG is unknown to us. Therefore,
+   * this method cannot be used to emit field lineage information.
+   *
+   * @param fieldOperations the operations to be recorded
+   * @deprecated use {@link #record(Collection)} to emit field operations for actions, that method requires the input
+   *             field to contain the previous originated operation name. Using that DAG can be computed so the field
+   *             linage can be computed.
+   */
+  @Override
+  @Deprecated
+  default void record(List<FieldOperation> fieldOperations) {
+    throw new UnsupportedOperationException("Lineage recording is not supported.");
+  }
+
+  /**
+   * Register lineage for this Spark program using the given reference name
+   *
+   * @param referenceName reference name used for source
+   *
+   * @param accessType the access type of the lineage
+   * @throws DatasetManagementException thrown if there was an error in creating reference dataset
+   * @throws TransactionFailureException thrown if there was an error while fetching the dataset to register usage
+   */
+  void registerLineage(String referenceName,
+                       String accessType) throws DatasetManagementException, TransactionFailureException;
```

Comment:
its better if this takes an enum rather than a string, otherwise the API needs to define what happens on invalid input.

Reply:
The `AccessType` enum is in data-fabric but that is not accessible from etl-api. And I don't think it is good to move the class to cdap-api, is there a good way to deal with it?

Reply:
you can create a separate enum class in cdap-etl-api. The implementation will translate it to the one in data-fabric.

I'm also not sure if we need to include READ_WRITE in this enum. Can you think of situations where it would be valuable?

Reply:
+1 to have a separate enum class. 
I have READ_WRITE in the implementation only to match with the existing AccessType, I can't think of when it is useful, will remove it in the new enum class.